rai-sdk/src/main/java/com/relationalai/Client.java

```diff
@@ -705,25 +705,22 @@ public TransactionAsyncResult execute(
             String source, boolean readonly,
             Map<String, String> inputs) throws HttpError, IOException, InterruptedException {
 
-        var id = executeAsync(database, engine, source, readonly, inputs).transaction.id;
+        var rsp = executeAsync(database, engine, source, readonly, inputs);
 
-        var transaction = getTransaction(id).transaction;
+        if (rsp.gotCompletedResult)
+                return rsp;
 
+        var transaction = getTransaction(rsp.transaction.id).transaction;
         while ( !("COMPLETED".equals(transaction.state) || "ABORTED".equals(transaction.state)) ) {
```

Comment:
don't we have to poll until it's either completed or aborted? like this https://github.com/RelationalAI/rai-sdk-go/blob/bbc1c3bba86b43865e983d992d48161e5652530f/rai/client.go#L1270

Reply:
I hope we can get to a place where the polling loop looks pretty much exactly the same between SDKs

Reply:
> don't we have to poll until it's either completed or aborted?

I'm confused, @vilterp, that looks like exactly what this loop is doing.

Reply:
Derp, don’t know how I missed that. My bad